packages/story-editor/src/components/toggleButton/index.js

```diff
@@ -35,57 +39,135 @@ import { forwardRef } from '@googleforcreators/react';
+  ${({ isOpen, hasText, $size = buttonSize, theme }) => css`
+    height: ${resolveSizeUnits($size)};
+    min-width: ${resolveSizeUnits($size)};
+    color: ${theme.colors.fg.primary};
+    border-color: ${
+      isOpen ? theme.colors.bg.secondary : theme.colors.border.defaultNormal
+    };
+    background-color: ${
+      isOpen ? theme.colors.bg.secondary : theme.colors.bg.primary
+    };
+    ${
+      hasText &&
+      css`
+        ${themeHelpers.expandPresetStyles({
+          preset: {
+            ...theme.typography.presets.paragraph[
+              THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL
+            ],
+          },
+          theme,
+        })};
+
+        .badge {
+          margin: 0 8px;
+        }
+      `
+    }
+
+    .badge.open,
+    &:hover .badge,
+    &:focus .badge {
+      background-color: ${theme.colors.bg.quaternary};
+    }
+
+  }`}
+
+  .badge {
+    transition: background-color ${BUTTON_TRANSITION_TIMING};
+  }
+
+  // Margin is set to -8px to compensate for empty space
+  // around the actual icon graphic in the svg
+  svg.main-icon {
+    height: 32px;
+    width: auto;
+    margin: -8px;
+    display: block;
+  }
 `;
 Button.propTypes = {
   hasText: PropTypes.bool,
   isOpen: PropTypes.bool,
+  $size: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
 };
 
-const Wrapper = styled.div`
-  width: 100%;
-  height: 100%;
-  display: flex;
-  justify-content: space-around;
-  align-items: center;
-`;
+// `badgeSize` is the default height and min-width of the badge circle/oval
+// Hard-coded here, but can be overridden by `$size` prop in the component
+const badgeSize = 22;
+
+// Defaults for badge come from typography.presets.label.small
+const badgeText =
+  dsTheme.typography.presets.label[
+    THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL
+  ];
 
-const NotificationCount = styled(Text).attrs({ as: 'span' })`
-  margin: -2px 0 -2px;
-  padding-right: 6px;
-  line-height: 20px;
-  color: ${({ theme }) => theme.colors.violet[20]};
+// TODO: Extract `CountBadge` to its own component
+// Props prefixed with `$` won't leak into element attributes
+// (`theme` is handled internally by Styled Components)
+// padding and border-radius properties use css `calc()` so units
+// other than px can be used for $size and still display correctly
+const CountBadge = styled(Text).attrs({ as: 'span', className: 'badge' })`
+  ${({ $size = badgeSize, $fontSize = badgeText.size, theme }) => css`
+    min-width: ${resolveSizeUnits($size)};
+    width: auto;
+    height: ${resolveSizeUnits($size)};
+    padding: 0 calc(${resolveSizeUnits($size)} / 4);
+    margin-left: 10px;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    color: ${theme.colors.fg.primary};
+    background-color: ${theme.colors.bg.tertiary};
+    border-radius: calc(${resolveSizeUnits($size)} / 2);
+    font-size: ${resolveSizeUnits($fontSize)};
+    line-height: 0;
+  `}
 `;
```

Comment:
This allows use of non-pixel values to be used. If the argument is numeric (either string or number), the value will be returned with `px` appended, where a value like `100%` will be returned as-is. This is similar to React's behavior of appending `px` to numbers used as values of `style` objects used for inline styling. In the `CountBadge` component, it allows use of `calc` with non-px values.

Reply:
Is there an example here where this is used, i.e. something like `100%` (or another non-numeric arg) is being passed?

Jus trying to better understand this since this introduces a new pattern to the code base.

Reply:
🤔 I'll do some soul searching (aka try to remember my original intention).

Reply:
Updated to use only 'px' values.